### ISSUE TITLE
Format array params

### ISSUE THREAD

**Issue by xr0m3oz** [dkrnl/SimpleXMLReader#5]
`<offer available="false" id="168828803">
   <categoryId>11293</categoryId>
   <param name="Brand">XiaoMi</param>
   <param name="Type">4G Smartphone</param>
   <param name="OS">Android 5.1</param>
</offer>
`

Can i get <param> attribute in view 
key_name=>value

Now i can get
`				foreach ($offer->param as $key=> $item) {
					echo "name:".$item."<br>";
				}
`
[0]=>XiaoMi
[1]=>4G Smartphone

Can i get 

[Brand]=>XiaoMi
[Type]=>4G Smartphone

**Comment by dkrnl** [comment]
Use _SimpleXMLElement::attributes()_ method: http://php.net/manual/en/simplexmlelement.attributes.php
foreach ($offer->param-**>attributes()** as $key=> $item)

**Comment by xr0m3oz** [comment]
Атрибуты искомого элемента получаются в нормальном виде с Ключ:значение.
`
SimpleXMLElement Object
(
    [@attributes] => Array
        (
            [available] => true
            [id] => 125800401
        )

    [categoryId] => 11312
    [currencyId] => USD
    [delivery] => true
    [description] => Geneva Non - functioning Sub - dials Ladies Quartz Watch with Plastic Band
    [goodsNumber] => 999
    [goodsStatus] => New
    [local_delivery_cost] => 0
    [manufacturer_warranty] => false
    [modified_time] => 1458658728
    [name] => Geneva Non - functioning Sub - dials Ladies Quartz Watch with Plastic Band
    [oldprice] => 10.85
    [oriCountry] => CN
    [param] => Array
        (
            [0] => Geneva
            [1] => Female table
            [2] => White
            [3] => Diamond
            [4] => Quartz watch
            [5] => Round
            [6] => Analog
            [7] => Alloy
            [8] => Plastic
            [9] => Folding clasp with safety
            [10] => Decorating small sub-dials
            [11] => 1.0 cm / 0.4 inches
            [12] => 4.2 cm / 1.65 inches
            [13] => 2.0 cm / 0.79 inches
            [14] => 0.063 kg
            [15] => 0.113 kg
            [16] => 14 x 4.2 x 1 cm / 5.50 x 1.65 x 0.39 inches
            [17] => 15 x 5.2 x 2 cm / 5.90 x 2.04 x 0.79 inches
            [18] => 1 x Geneva Watch
        )

    [pickup] => false
    [picture] => Array
        (
            [0] => http://gloimg.gearbest.com/gb/2015/201504/goods-img/1428950360886-P-2535778.jpg
            [1] => http://gloimg.gearbest.com/gb/2015/201504/goods-img/1428950361372-P-2535778.jpg
            [2] => http://gloimg.gearbest.com/gb/2015/201504/goods-img/1428950361737-P-2535778.jpg
            [3] => http://gloimg.gearbest.com/gb/2015/201504/goods-img/1428950361082-P-2535778.jpg
        )

    [postCountry] => CN
    [price] => 9.04
    [store] => false
    [topseller] => true
    [transportFee] => 0
    [transportTime] => 25
    [url] => https://ad.admitad.com/g/2316b8f85670d6392e6522af2ed61b/?i=5&ulp=http%3A%2F%2Fwww.gearbest.com%2Fwomen-s-watches%2Fpp_174857.html
)
`

`
<?xml version="1.0" encoding="UTF-8"?>
<offer available="false" id="168828803">
   <categoryId>11293</categoryId>
   <currencyId>USD</currencyId>
   <delivery>true</delivery>
   <description>XiaoMi Redmi 3 16GB ROM 4G Smartphone</description>
   <goodsNumber>0</goodsNumber>
   <goodsStatus>New</goodsStatus>
   <local_delivery_cost>0</local_delivery_cost>
   <manufacturer_warranty>false</manufacturer_warranty>
   <modified_time>1458658728</modified_time>
   <name>XiaoMi Redmi 3 16GB ROM 4G Smartphone</name>
   <oldprice>382.76</oldprice>
   <oriCountry>CN</oriCountry>
   <param name="Brand">XiaoMi</param>
   <param name="Type">4G Smartphone</param>
   <param name="OS">Android 5.1</param>
   <param name="Service Provide">Unlocked</param>
   <param name="Language">English, French, Spanish, Russian, German, Italian, Portuguese</param>
   <param name="SIM Card Slot">Dual SIM,Dual Standby</param>
   <param name="SIM Card Type">Micro SIM Card,Nano SIM Card</param>
   <param name="CPU">Qualcomm Snapdragon 616</param>
   <param name="Cores">Octa Core</param>
   <param name="GPU">Adreno-405</param>
   <param name="RAM">2GB RAM</param>
   <param name="ROM">16GB</param>
   <param name="External memory">TF card up to 128GB (not included)</param>
   <param name="Wireless Connectivity">3G,4G,A-GPS,Bluetooth,GPS,GSM,WiFi</param>
   <param name="WIFI">802.11a/b/g/n/ac wireless internet</param>
   <param name="Network type">FDD-LTE+WCDMA+GSM</param>
   <param name="2G">GSM 850/900/1800/1900MHz</param>
   <param name="3G">WCDMA 850/900/1900/2100MHz</param>
   <param name="4G">FDD-LTE 1800/2100/2600MHz</param>
   <param name="Screen type">Capacitive (5-Points)</param>
   <param name="Screen size">5.0 inch</param>
   <param name="Screen resolution">1280 x 720 (HD 720)</param>
   <param name="Camera type">Dual cameras (one front one back)</param>
   <param name="Back camera">13.0MP,with flash light and AF</param>
   <param name="Front camera">5.0MP</param>
   <param name="Video recording">Yes</param>
   <param name="Aperture">f/2.0</param>
   <param name="Touch Focus">Yes</param>
   <param name="Auto Focus">Yes</param>
   <param name="Flashlight">Yes</param>
   <param name="Camera Functions">Face Detection,HDR,Panorama Shot</param>
   <param name="Picture format">BMP,GIF,JPEG,PNG</param>
   <param name="Music format">AAC,AMR,MP3,WAV</param>
   <param name="Video format">ASF,AVI,MKV,MP4</param>
   <param name="MS Office format">Excel,PPT,Word</param>
   <param name="E-book format">PDF,TXT</param>
   <param name="Live wallpaper support">Yes</param>
   <param name="Games">Android APK</param>
   <param name="I/O Interface">3.5mm Audio Out Port,Micro USB Slot,TF/Micro SD Card Slot</param>
   <param name="Sensor">Ambient Light Sensor,E-Compass,Gravity Sensor,Proximity Sensor</param>
   <param name="Notification LED">Yes</param>
   <param name="Sound Recorder">Yes</param>
   <param name="Additional Features">3G,4G,Alarm,Bluetooth,Browser,Calculator,Calendar,E-book,GPS,MP3,MP4,People,Sound Recorder,Video Call,Wi-Fi</param>
   <param name="Battery Capacity (mAh)">Built-in 4100mAh</param>
   <param name="Battery Type">Lithium-ion Polymer Battery</param>
   <param name="Cell Phone">1</param>
   <param name="Power Adapter">1</param>
   <param name="USB Cable">1</param>
   <param name="SIM Needle">1</param>
   <param name="Product size">13.93 x 6.96 x 0.85 cm / 5.48 x 2.74 x 0.33 inches</param>
   <param name="Package size">18.00 x 12.00 x 6.00 cm / 7.09 x 4.72 x 2.36 inches</param>
   <param name="Product weight">0.144 kg</param>
   <param name="Package weight">0.500 kg</param>
   <pickup>false</pickup>
   <postCountry>CN</postCountry>
   <price>141.07</price>
   <store>false</store>
   <topseller>true</topseller>
   <transportFee>0</transportFee>
   <transportTime>25</transportTime>
   <url>https://ad.admitad.com/g/2316b8f85670d6392e6522af2ed61b/?i=5&amp;ulp=http%3A%2F%2Fwww.gearbest.com%2Fcell-phones%2Fpp_298117.html</url>
   <vendor>Xiaomi</vendor>
   <youtube>https://www.youtube.com/watch?v=guGZbnsCKB4</youtube>
</offer>
`
Меня интересует как можно сделать тут  [param] => Array ассоциативный массив.

**Comment by dkrnl** [comment]
```php
$params = iterator_to_array($offer->param);
$params = array_combine(array_map(function($i) {
    return (string)$i->name;
}, $params), array_map(function($k, $v) {
    return (string)$i;
}, $params));
```
